### DISCUSSION TITLE
1.0 Roadmap

### DISCUSSION THREAD

**Discussion by p1c2u** [p1c2u/openapi-core#492] [Ideas]
# Roadmap
* Files (binary request/response) support ✅ #710 #646 #694 #678
* Stable public API ☑️#788
* Remove deprecations ✅ 
* Documentation ☑️#788

**Comment by dtook-nbs** [comment]
Hello.  @p1c2u  Thank you for your work here.  Do you know when v1.0 stable RC will be available?  Is there any date in the future such as end of 2023 or end of 2024?  Thank you.

**Comment by p1c2u** [comment]
Hi @dtook-nbs 

that's a good question. Let me answer the question by giving the update of the roadmap:

* Documentation is in progress (types, documentation draft, docs workflow are done) and I would say it's 50% done. Still need to include docstrings and examples.
* Stable public API is basically done with the new release (0.17.x)
* Deprecations will be removed with the following release (0.18.x)

As you can see it's not so far (2-3 new releases) and personally I put deadline for mid 2023.

**Comment by dtook-nbs** [comment]
Hi there.  Wow that is amazing.  Thanks for getting back to me.  And thanks for your hard work.

**Comment by dtook** [comment]
Hello @p1c2u 

Is there any update on the v1 roadmap?  I know you said mid 2023 before.  

Thank you!

**Reply by p1c2u** [reply]
Documentations was delayed because of additional work that needed to be done with jsonschema 4.18 upgrade

**Comment by p1c2u** [comment]
One important point added: Files (binary request/response) support
